### PR TITLE
#631 base image

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
#632 themed and svg icons
#638 Inline SVG

New optional properties in Icon
- light_path: str
- dark_path: str

### CODE REVIEW

**Review by dinhlongviolin1 — APPROVED** [review]

File: src/taipy/gui/icon.py [hunk 1]
```diff
@@ -45,14 +46,29 @@ def __init__(self, path: str, text: t.Optional[str] = None) -> None:
                 the application directory.
             text (Optional[str]): The text associated to the image. If _text_ is None,
                 there is no text associated to this image.
+            light_path (Optional[str]): The path to the light theme image (fallback to _path_ if not defined).
+            dark_path (Optional[str]): The path to the dark theme image (fallback to _path_ if not defined).
+            svg (Optional[bool]): Indicates that the _path_ should be treated as an SVG resource (defaults to True if _path_ ends with .svg).
```
Comment (FabienLelaquais) [inline src/taipy/gui/icon.py]: Where is this doc ("defaults to True if _path_ ends with .svg") implemented?
What if path is svg, but light_path is not?

File: src/taipy/gui/icon.py [hunk 2]
```diff
@@ -45,14 +46,29 @@ def __init__(self, path: str, text: t.Optional[str] = None) -> None:
                 the application directory.
             text (Optional[str]): The text associated to the image. If _text_ is None,
                 there is no text associated to this image.
+            light_path (Optional[str]): The path to the light theme image (fallback to _path_ if not defined).
```
Comment (FabienLelaquais) [inline src/taipy/gui/icon.py]: Please use '*' for parameter name references instead of '_'.

File: gui/src/components/Taipy/Image.tsx [hunk 3]
```diff
@@ -129,25 +129,35 @@ const Image = (props: ImageProps) => {
...
+                                className={getSuffixedClassNames(className, "-label")}
+                            >
+                                {label}
+                                <ImageMarked className="MuiImageMarked-root" />
+                            </Typography>
+                        </ImageSpan>
+                    )}
+                </ImageButton>
+            ) : (
+                <img id={id} src={content} style={style} className={className} />
```
Comment (FabienLelaquais) [inline gui/src/components/Taipy/Image.tsx]: So the potential svg is embedded in an img... hmm ok...

**Review by FabienLelaquais — APPROVED** [review]
Nice job!

File: src/taipy/gui/icon.py (hunk 1 above)
Comment (FredLL-Avaiga) [reply src/taipy/gui/icon.py]: In [icon.tsx:57](https://github.com/Avaiga/taipy-gui/blob/feature/%23631-%23632-base-image-and-themed-icons/gui/src/utils/icon.tsx#L57)
the svg property is meant to allow non svg ending files to be treated as SVG ... Is it really necessary?

File: src/taipy/gui/icon.py (hunk 2 above)
Comment (FredLL-Avaiga) [reply src/taipy/gui/icon.py]: will do

File: gui/src/components/Taipy/Image.tsx (hunk 3 above)
Comment (FredLL-Avaiga) [reply gui/src/components/Taipy/Image.tsx]: Didn't think of this one ...

Comment (ghost) [reply gui/src/components/Taipy/Image.tsx]: Yeah I was not sure I read it well, but it tickled me as well...

Comment (FredLL-Avaiga) [reply gui/src/components/Taipy/Image.tsx]: We could also handle content as svg text ie 
```
content = "<SVG> ...</SVG>"
<|image|content={content}|>
```
What do you think ?

Comment (ghost) [reply gui/src/components/Taipy/Image.tsx]: You meam, right in a GUI app, on the user side ? It's a bit dirty, don't you think ? SVG code can be messy, and putting that in your app's code may make it look like crap.

Comment (FredLL-Avaiga) [reply gui/src/components/Taipy/Image.tsx]: just another option ... it can be loaded from a file or returned by a function ...

Comment (ghost) [reply gui/src/components/Taipy/Image.tsx]: OK, why not, then.

Comment (FredLL-Avaiga) [reply gui/src/components/Taipy/Image.tsx]: one could use this to insert svg generated from template

File: src/taipy/gui/icon.py (hunk 1 above)
Comment (FabienLelaquais) [reply src/taipy/gui/icon.py]: Oh right.
And as you say... we may not even need this.

File: gui/src/components/Taipy/Image.tsx (hunk 3 above)
Comment (FabienLelaquais) [reply gui/src/components/Taipy/Image.tsx]: ... and animate it!

**Pull request merged**